### PR TITLE
Added support for asymmetric input during training

### PR DESCRIPTION
**Pull request by Mward94**
Adding support for training a model using asymmetric input.
I've updated the existing transform code to be consistent with the usage of X/Y, height/width.
I've also decoupled the scale into height and width components and updated the code respectively.

I've tested this on the MPII dataset as input, and visually observed that the target heatmaps correctly line up with the input images for both symmetric and asymmetric input. Additionally, I've tested leaving the input resolution as it was (256x256), and observed that the heatmaps again are as expected.

Two notes:

- The MPII dataloader need to know the output heatmap resolution to construct the target heatmaps. Historically, this was given as an argument to the constructor, but likely was never manually specified. The output resolution is driven by the striding of the model, hence there should only be 1x valid output resolution per-input resolution. I've currently defined a constant to reflect the ratio between input/output resolution, however there may be other ways to solve this. One method would be to let the user specify the output resolution (or perhaps even the input-output resolution relationship).

- On line 116 of transforms.py, there was a check for a scale factor < 2. Given the scale is now decoupled between width and height, I've added a corresponding condition, however I'm uncertain what this is actually checking for, so I haven't tested if the replacement condition functions as expected.

### CODE REVIEW

File: src/stacked_hourglass/utils/transforms.py [hunk 1]
```diff
@@ -78,38 +86,52 @@ def transform(pt, center, scale, res, invert=0, rot=0):
...
 
 
 def crop(img, center, scale, res, rot=0):
+    """
+    img: torch.Tensor: shape: C,H,W
+    center: torch.Tensor: shape: x, y
+    scale: [H/200, W/200]
+    res: H,W
+    """
+    # If given 1D scale as a torch Tensor, convert to a pair of scales
```
Comment (anibali) [inline src/stacked_hourglass/utils/transforms.py]: I'm worried about the robustness of this code. How about:

```python
scale = np.asarray(scale).flatten()
if len(scale) == 1:
    scale = scale.repeat(2)
```

File: src/stacked_hourglass/utils/transforms.py [hunk 2]
```diff
@@ -41,12 +41,20 @@ def get_transform(center, scale, res, rot=0):
     """
     General image processing functions
     """
+    # If given 1D scale as a torch Tensor, convert to a pair of scales
```
Comment (anibali) [inline src/stacked_hourglass/utils/transforms.py]: See https://github.com/anibali/pytorch-stacked-hourglass/pull/3/files#r500003349

**Comment by anibali** [comment]
Thanks for that, it will be great to finally have proper support for asymmetric input. If you could just address my little nitpick I'll be glad to merge it :+1:

File: src/stacked_hourglass/utils/transforms.py (hunk 1 above)
Comment (Mward94) [reply src/stacked_hourglass/utils/transforms.py]: That's a great suggestion! This will be a nice robust solution

**Pull request merged**